### PR TITLE
TDL-15251: Add required fields for age_gender_audience_report

### PR DESCRIPTION
**Pull request by namrata270998**
# Description of change
- Updated the required columns dictionary by adding required fields for AgeGederAudienceReport

# Manual QA steps
 - Run the discovery mode and check that the catalog has the fields AccountName, AdGroupName, AgeGroup, Gender marked as inclusion automatic or not.

# Risks
 - 
 
# Rollback steps
 - revert this branch

### CODE REVIEW

**Review by dbshah1212 — APPROVED** [review]

**Review by savan-chovatiya — APPROVED** [review]

File: tap_bing_ads/reports.py [hunk 1]
```diff
@@ -32,7 +32,13 @@
         'Ctr',
         'Impressions'
     ],
-    'GoalsAndFunnelsReport': ['Goal']
+    'GoalsAndFunnelsReport': ['Goal'],
+    'AgeGenderAudienceReport': [
```
Comment (karanpanchal-crest) [inline tap_bing_ads/reports.py:36]: @namrata270998 Can you please provide a comment above this code change?

**Review by karanpanchal-crest — CHANGES_REQUESTED** [review]

File: tap_bing_ads/reports.py [hunk 2]
```diff
@@ -32,7 +32,13 @@
...
+        'AccountName',
+        'AdGroupName',
+        'AgeGroup',
+        'Gender'
```
Comment (karanpanchal-crest) [inline tap_bing_ads/reports.py:40]: @namrata270998 From the bing docs, I also see 'TimePeriod' as the required key. However, I don't see the same mentioned in this list

**Review by karanpanchal-crest — CHANGES_REQUESTED** [review]

File: tap_bing_ads/reports.py (hunk 2 above)
Comment (namrata270998) [reply tap_bing_ads/reports.py:40]: Yes, that's correct. However, it is also a common field for all the reports, hence it is added in another `REPORT_REQUIRED_FIELDS ` list which can be found in this line
https://github.com/singer-io/tap-bing-ads/pull/86/files#diff-5215cbddcb33118eb2c3a413c2ded5a5a94bc52f7b1c7fa88131ba2dcba85684R14

File: tap_bing_ads/reports.py (hunk 1 above)
Comment (namrata270998) [reply tap_bing_ads/reports.py:37]: Added comments

**Review by karanpanchal-crest — APPROVED** [review]

**Review by KrisPersonal — APPROVED** [review]

File: tests/test_discovery.py [hunk 3]
```diff
@@ -129,8 +129,12 @@ def test_run(self):
                 expected_foreign_keys = self.expected_foreign_keys()[stream]
                 expected_replication_keys = self.expected_replication_keys()[stream]
                 expected_required_keys = self.expected_required_fields()[stream]
+                # adding '_sdc_report_datetime in the expected automatic fields list as there was 
+                # a failure experienced for all the streams that the particular field was missing
+                # in the expected fields.
                 expected_automatic_fields = expected_primary_keys | expected_replication_keys \
-                    | expected_foreign_keys | expected_required_keys
+                    | expected_foreign_keys | expected_required_keys | {'_sdc_report_datetime'}
```
Comment (kspeer825) [inline tests/test_discovery.py:136]: The assertion for this is not executing.

**Review by kspeer825 — CHANGES_REQUESTED** [review]

File: tests/test_discovery.py (hunk 3 above)
Comment (namrata270998) [reply tests/test_discovery.py:136]: added the missing assertions

**Review by kspeer825 — APPROVED** [review]

**Pull request merged**